corehq/apps/export/views.py

```diff
@@ -632,19 +632,21 @@ def _get_download_task(self, in_data):
             )
         return download
 
+    def _get_and_rebuild_export_schema(self, export_id):
+        export_object = self.get_export_schema(self.domain, export_id)
+        export_object.update_schema()
+        return export_object
+
     def _get_bulk_download_task(self, export_specs, export_filter):
+        for export_spec in export_specs:
+            export_id = export_spec['export_id']
+            self._get_and_rebuild_export_schema(export_id)
         export_helper = CustomBulkExportHelper(domain=self.domain)
         return export_helper.get_download_task(export_specs, export_filter)
 
     def _get_single_export_download_task(self, export_spec, export_filter, max_column_size=2000):
-        try:
-            export_data = export_spec
-            export_object = self.get_export_schema(self.domain, export_data['export_id'])
-        except (KeyError, IndexError):
-            raise ExportAsyncException(
-                _("You need to pass a list of at least one export schema.")
-            )
-        export_object.update_schema()
+        export_id = export_spec['export_id']
+        export_object = self._get_and_rebuild_export_schema(export_id)
```

Comment:
interesting. I was expecting this to catch a `SchemaMismatchException` somewhere and filing off an async task. but I guess that happens somewhere inside `update_schema`?